pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -414,11 +414,19 @@ func generateOriginalContainerRuntimeConfigs(templateDir string, cc *mcfgv1.Cont
 			break
 		}
 	}
-	if errStorage != nil || errCRIO != nil || errRegistries != nil {
-		return nil, nil, nil, fmt.Errorf("could not generate old container runtime configs: %v, %v, %v", errStorage, errCRIO, errRegistries)
+	// Find Policy JSON
+	for _, gmc := range generatedConfigs {
+		config, errPolicy = findPolicyJSON(gmc)
+		if errPolicy == nil {
+			gmcPolicyJSON = config
+			break
+		}
+	}
+	if errStorage != nil || errCRIO != nil || errRegistries != nil || errPolicy != nil {
```

Comment:
Non-blocking: This repetitive code doing 4 times the same two nested loops with different loops, with 4 sets of variables, 4 sets of errors, and 4 sets of return values (which are never used at the same time), is about at the point of being too ugly.

A single piece of code building a generic `Map<filename string, ignore *igntypes.File>`, and the callers just indexing into that with the necessary path, and handling failures separately, would be quite a bit shorter and somewhat more efficient.

Perhaps even make the function extract `contents []byte` for all of those files; that would do a fair amount of unnecessary data conversions to simplify the consumers further.

(And, going _way_ beyond the scope of this PR, it intuitively seems to me that that inefficiency could be best avoided by passing the required file names down into `GenerateMachineConfigsForRole` so that the unnecessary data is not generated in the first place, instead of only not converting it.)

Reply:
Will fix this up in a following PR